components/src/core/Drawer/Drawer.tsx

```diff
@@ -26,18 +26,112 @@ const useStyles = makeStyles((theme: Theme) =>
     })
 );
 
+// type shared by Drawer, DrawerBody, DrawerNavGroup, NestedNavItem
+// these types are inherited from the Drawer level to the NestedNavItem
+// parent props will be overriden by the child props if defined
+export type PXBlueDrawerInheritableProperties = {
+    // Background color for the 'active' item
+    activeItemBackgroundColor?: string;
+
+    // shape of the active item background
+    activeItemBackgroundShape?: 'round' | 'square';
+
+    // Font color for the 'active' item
+    activeItemFontColor?: string;
+
+    // Icon color for the 'active' item
+    activeItemIconColor?: string;
+
+    // Whether to have chevrons for all menu items
+    chevron?: boolean;
+
+    // Icon used to collapse drawer
+    // default is expandIcon rotated 180 degrees
+    collapseIcon?: JSX.Element;
+
+    // Whether to show a line between all items
+    divider?: boolean;
+
+    // Icon used to expand drawer
+    expandIcon?: JSX.Element;
+
+    // Whether to hide the paddings reserved for menu item icons
+    hidePadding?: boolean;
+
+    // The color used for the item text
+    itemFontColor?: string;
+
+    // The color used for the icon
+    itemIconColor?: string;
+
+    // internal API
+    // will apply to all menu items when onClick
+    onItemSelect?: () => void;
+
+    // Whether to apply material ripple effect to items
+    ripple?: boolean;
+};
+
+// type shared by Drawer, DrawerBody, DrawerNavGroup
+// inheritable props but not for NestedNavItem
+export type PXBlueDrawerInheritableGroupProperties = {
+    // itemID for the 'active' item
+    activeItem?: string;
+
+    // background color for nested menu items
+    nestedBackgroundColor?: string;
+
+    // Whether to show a line between nested menu items
+    nestedDivider?: boolean;
+
+    // internal API (public only for Drawer)
+    // Whether the group is expanded
+    open?: boolean;
+
+    // Font color for group header
+    titleColor?: string;
+} & PXBlueDrawerInheritableProperties;
+
 export type DrawerComponentProps = {
-    open: boolean;
+    // drawer width
     width?: number;
-} & Omit<DrawerProps, 'translate'>;
+
+    // types of drawer
+    // only open or variant is allowed but not both
+    variant?: 'permanent' | 'persistent' | 'temporary';
```

Comment:
1) this could just be inherited from DrawerProps (MUI)
2) we shouldn't allow this until we've actually implemented the different behaviors for the variants

Reply:
Let me do that in a separate PR